go/hamming/hamming.go

```diff
@@ -1,4 +1,37 @@
+// This package implements an hamming distance calculation function
+// aimed at DNA strands
+
 package hamming
 
-func Distance(a, b string) (int, error) {
+import (
+	"errors"
+	"strings"
+)
+
+/*  Distance returns the hamming distance between two dna strands
+    passed as string parameters. Returns -1 if the strands are of
+	different lengths.
+    parameters :
+		a : strand as a string
+		b : strand as a string
+*/
+func Distance(a, b string) (distance int, distanceError error) {
+	if len(a) != len(b) {
+		distance = -1
+		distanceError = errors.New("Distance expects strands to be of equal lengths.")
+	} else {
+		distanceError = nil
+		distance = 0
+		var aReader = strings.NewReader(a)
```

Comment:
Given that the DNA is only composed of 4 different letters, I don't think Rune reading is really necessary. It doesn't hurt, but just taking the bytes of the strings (`a[i]` kind of thing) should be fine as well.

Reply:
Completely forgot we were dealing about DNA, silly me. applying your recommendartions